src/core/gui/toolbarMenubar/ToolButton.cpp

```diff
@@ -52,9 +50,25 @@ auto ToolButton::createItem(bool horizontal) -> GtkToolItem* {
     GtkToolItem* it = gtk_tool_item_new();
     if (popover) {
         GtkMenuButton* menubutton = GTK_MENU_BUTTON(gtk_menu_button_new());
+        gtk_widget_set_can_focus(GTK_WIDGET(menubutton), false);  // todo(gtk4) not necessary anymore
         gtk_menu_button_set_popover(menubutton, popover.get());
         gtk_menu_button_set_direction(menubutton, GTK_ARROW_DOWN);
 
+        if (target && toggle) {
+            // Pressing the menubutton also selects the tool
```

Comment:
Looks like this has a serious downside with this: When the selection tool is active and you press the menubutton for the pen to change the line style, the selection is lost and you can't change the line style of the selection.

Reply:
Yes. I've been looking for a good solution, but it's not clear yet.

Reply:
The toolbar can have individual buttons to select the linestyle
![Screenshot_20231013_181541](https://github.com/xournalpp/xournalpp/assets/17818041/0b2b6014-67f4-4b91-96cf-f55ac8ad3f6a)

What behaviour do we want of them?

1. Always sensitive. Change the linestyle if the current tool is PEN, change the selection's line style if something is selected, change the current tool to PEN otherwise.
2. Change the linestyle if the current tool is PEN, change the selection's line style if something is selected, and disabled otherwise.

As far as I can tell, there are 3 places where the line style can be selected: in the menu (behaviour 1. is better, I think), in the dropdown popover of the toolbar PEN button (behaviour 1. is better, I think) and those individual line style selectors (1. or 2., I don't know)

Reply:
With "change the current tool to PEN otherwise" you probably mean "change the current tool to PEN and set the line style according to the button/menu".

In general I think having sensitive widgets that can do something useful is better than having insensitive widgets (that are still visible). So I would go with option 1 for all 3 places.

Reply:
I'm a bit scared we'll get a bunch of "Setting dashed line on highlighter does not work". I'd say that if we use 1, then the icon should make it clear this will activate the pen. It'd be like a "tool button with specific options" that has been asked somewhere in an issue.

Reply:
Ok, fair point, the action should be intuitively understandable, which may not be the case for the individual line style selectors changing to the pen tool. To create "tool buttons with specific options" icons (like in https://github.com/xournalpp/xournalpp/issues/2431#issuecomment-820060571) looks out of scope for this PR though. So then I change my opinion and recommend to disable the individual line style selectors if the current tool is not the PEN and nothing is selected, at least if it doesn't make things too complicated.

Reply:
TBH it's simpler to create a bunch of icons with a pen on top of the linestyle and use those for the standalone linestyle buttons. It's cleaner than the solution I have in mind for implementing 2.

Reply:
Like that: 
![Screenshot_20231013_205153](https://github.com/xournalpp/xournalpp/assets/17818041/9b5d8783-3be1-42cd-bbe4-37316b7f745c)

Reply:
Ok, looks good.

Reply:
And the Lucide ones
![Screenshot_20231013_210241](https://github.com/xournalpp/xournalpp/assets/17818041/c7ddcad7-17cb-4d61-a5ea-46f58c96bafe)